### DISCUSSION TITLE
Newest Version Causes Field Issues?

### DISCUSSION THREAD

**Discussion by CosmasSC128** [Budibase/budibase#5964] [Help] [answered]
Trying to display information in a form field as a default value. This worked in the last version without problems.

Newest version means that same code will display wonderfully in a paragraph widget, or as a placeholder value, but not in a form field default value.

Picture attached of the string displayed in a paragraph, then successfully displayed as a 'placeholder' but failing to display as a 'default value'.
![Screenshot 2022-05-18 at 14 50 34](https://user-images.githubusercontent.com/93920520/169059668-103d0893-5311-4967-9c2d-b47687873446.png)

One more note, using handlebars does not work either, though it displays fine for a paragraph widget as well, or a placeholder on a form field as well.

Finally, the data is passed through perfectly to a query for example, meaning the data is accessible, and is being accurately referenced, but for some reason form fields (default value) is no longer behaving properly. Or at least as it was.

If anyone has any ideas, please let me know.

**Comment by melohagan** [comment, marked as answer]
Hi @CosmasSC128 

Do you know the version numbers you have upgraded from and upgraded to? 

Are you trying to dynamically update the default value, or is it not even setting the value on initial load of the page?

**Reply by CosmasSC128** [reply]
To 1.0.158  from 1.0.99.

**Reply by melohagan** [reply]
Thank you. Within that time frame a change was made to the default value behavior: https://github.com/Budibase/budibase/issues/5392

It used to change dynamically, but was changed to only default on load of the page. 
An On Change action has been added though so you can dynamically update values that way: https://docs.budibase.com/docs/actions#update-field-value

**Reply by CosmasSC128** [reply]
I want some form field values to change based on information a form value set. Aka UploadForm.Value (.FieldName of some kind etc). When I change something on the form, information in the value set changes, and I'd like the form fields to update based on that. Is that possible? It doesn't seem obvious from the linked material.

**Reply by melohagan** [reply]
Hi @CosmasSC128 You can do that.

1. Select the field you want to be the trigger, and click 'On Change'
2. Add the 'Update Field Value' event for each form field you want to set.

**Reply by CosmasSC128** [reply]
Can something other than a field be the trigger to update another field? I was hoping to update fields on a form based on an attachment added to the attachment widget.

**Reply by melohagan** [reply]
What you might be able to do is have a button with an 'Upload to S3' action as well as the update field actions.

**Reply by CosmasSC128** [reply]
There seems to be an issue where, I can update a field value by button or other means (on action value setting), but that change in value will no trigger that fields "on change" behaviour.

Example Field A  has "on change", that if it is changed, it changes Field B's value. I use a button to change Field A's value, then Field B will not change. If I manually update Field A, it will change Field B.

Is there a way to fix this or get around it?

**Reply by CosmasSC128** [reply]
Thank you for all the help by the way.

**Reply by CosmasSC128** [reply]
"On Change" Actions for fields seem to occur when I merely click in and out of them, and not at all related to if the actual value changes or not. If that wasn't clear.

**Reply by melohagan** [reply]
For text/number fields the on change is triggered when the element looses focus, e.g. you click off or press tab.

**Reply by CosmasSC128** [reply]
Is there another way to trigger on change?

**Reply by melohagan** [reply]
Triggers only apply to fields and buttons at the moment. If there is a particular trigger you would like, feel free to request that feature.

**Reply by CosmasSC128** [reply]
Please add a trigger for a value of a field or paragraph changing, and not just when triggered by human interaction. Thank you.
